### ISSUE TITLE
Diffuse radiation shading

### ISSUE THREAD

**Issue by Mathadon** [open-ideas/IDEAS#735]
Shading models currently do not affect the value of `iSolDif`, the diffuse solar irradiation. This may however be the dominant source of solar energy when all direct solar irradiation is shaded. It would therefore make sense to implement diffuse solar shading effects.

There are two sources of diffuse solar irradiation. Irradiation from the ground and irradiation from the sky. These two sources are summed before passing them into the shading models. It may need to be able to treat these variables separately. Therefore I suggest to change the implementation such that both variables are passed to the shading models.

**Comment by Mathadon** [comment]
For current implementation I propose to assume that side fins, buildings and the lower side of overhangs are shaded, such that their diffuse solar irradiation is zero. It should however be doable to also take into account the amount of solar irradiation reflected by these objects.

**Comment by Mathadon** [comment]
In the box shading model we have

```
fraSun = max(0,1-(1-overhang.fraSun)-(1-sideFins.fraSun))
```

should this not be

```
fraSun = overhang.fraSun*sideFins.fraSun
```
i.e. the unshaded fraction equals the fraction that is unshaded by the overhang AND which is unshaded by the sideFins?

**Comment by Mathadon** [comment]
Before change:
![bestest](https://cloud.githubusercontent.com/assets/5970879/26510749/9e8ec5e2-4213-11e7-8f39-aab877b27cc4.png)

after change: 
![bestest](https://cloud.githubusercontent.com/assets/5970879/26510735/9344b692-4213-11e7-8c73-25dbd67d36c3.png)

**Comment by GlennReynders** [comment]
solved in #744
